pkg/image/importer/credentials.go

```diff
@@ -69,31 +67,50 @@ func (s *keyringCredentialStore) Basic(url *url.URL) (string, string) {
 	return basicCredentialsFromKeyring(s.DockerKeyring, url)
 }
 
-func NewCredentialsForSecrets(secrets []kapi.Secret) auth.CredentialStore {
-	return &secretCredentialStore{secrets: secrets}
+func NewCredentialsForSecrets(secrets []kapi.Secret) *SecretCredentialStore {
+	return &SecretCredentialStore{secrets: secrets}
 }
 
-type secretCredentialStore struct {
-	secrets []kapi.Secret
+func NewLazyCredentialsForSecrets(secretsFn func() ([]kapi.Secret, error)) *SecretCredentialStore {
+	return &SecretCredentialStore{secretsFn: secretsFn}
```

Comment:
Init already has to be lazy, because the keyring is expensive.  The behavior of this story is fundamentally lazy, it's really just that for tests and other scenarios where you already have secrets the laziness is obscuring intent.